### ISSUE TITLE
Allow using unit as an environment variable

### ISSUE THREAD

**Issue by tfausak** [dmjio/envy#13]
TL;DR: Can Envy add a `Var` instance for `()`? 

I want to keep all of my application configuration in a single data type. Most of the config comes from the environment, but some of it comes from other sources. (The stuff that comes from other sources is more like state than config, but it makes sense to me to keep it all together.) For example, I might read a [`Port`](https://www.stackage.org/haddock/lts-9.0/warp-3.2.13/Network-Wai-Handler-Warp.html#t:Port) from the environment but also have a [`Manager`](https://www.stackage.org/haddock/lts-9.0/warp-3.2.13/Network-Wai-Handler-Warp-Internal.html#t:Manager) in the config. 

Obviously the manager can't come from the environment. I wanted to express this in the type system, so I tried something like this:

``` hs
data Config manager = Config { port :: Port, manager :: manager }
  deriving Generic

type ConfigWithoutManager = Config () -- or Void

type ConfigWithManager = Config Manager

instance DefConfig ConfigWithoutManager where
  defConfig = Config { port = 8080, manager = () }

instance FromEnv ConfigWithoutManager
```

With that setup I figured I could [`decodeEnv`](https://www.stackage.org/haddock/lts-9.0/envy-1.3.0.2/System-Envy.html#v:decodeEnv) to get a `ConfigWithoutManager`, make the manager, then return a `ConfigWithManager`. Unfortunately the whole thing screeches to a halt with:

> No instance for `(Var ())`

So that approach doesn't work. I can work around it by using `Bool` instead of `()`, but that's not great. It would be nice if Envy could provide a `Var` instance for `()`. Obviously unit doesn't convey any information, so it doesn't make a ton of sense to use it in an environment variable. But it can be handy in situations like this.

**Comment by dmjio** [comment]
Ah, this is super interesting. Yea I don't see why not. Previously I've always just initialized the manager inside of `FromEnv` since there is a `MonadIO` instance.

```haskell
instance Var () where
  toVar = const "()"
  fromVar = const $ Just ()
```

**Comment by tfausak** [comment]
Ah, I didn't know about the `MonadIO` instance because the `Parser` data type isn't exported. Regardless, if you do that you'll have to use `undefined` (or `error` or `unsafePerformIO`) in `defConfig`.

**Comment by dmjio** [comment]
Right, if you use `liftIO` in `FromEnv`, the assumption is that you're no longer using generics. But I can definitely see a case where you'd like to parse the environment w/o initializing connection pools, and the `Var ()` approach seems pretty nice. Before, I'd have to do something like this.

```haskell
instance FromEnv Config where
  fromEnv = do
    Config <$> liftIO createConPool 
           <*> envMaybe "PG_PORT" .!= 5432
```
But again, I can see a case for not wanting to do this, and initializing the connection pool at a later stage.

**Comment by dmjio** [comment]
@tfausak, and yea, `Parser` the type should probably be exported.
